### ISSUE TITLE
loading BAM files into memory 

### ISSUE THREAD

**Issue by lhogstrom** [ahasfura/parallel_preseq#2]
@mukarramtahir 

Was looking through the BAM reader section of your code. I saw that your using samtools view to load in all of the lines of the code at ounce:
![image](https://cloud.githubusercontent.com/assets/3091474/11449583/46ffdf1e-9549-11e5-9456-720b69f44125.png)

This is fine when the files are small, but won't work on sequencing files that are hundreds of gigs. Do you want me to adapt this? I know you had mentioned that you might adapt this section to work in parallel anyway...

L

**Comment by mukarramtahir** [comment]
Yes, this is just a placeholder. I am going to work on this today, and will use index files to read the lines so that I don't overflow the memory when reading big files.
